pipelines/nlu-extension-build.yml

```diff
@@ -0,0 +1,74 @@
+trigger:
+  branches:
+    include:
+    - master
+  paths:
+    include:
+    - /extensions/
+    - /pipelines/extension-build.yml
+pr:
+  branches:
+    include:
+    - master
+  paths:
+    include:
+    - /extensions/
+    - /pipelines/extension-build.yml
+
+stages:
+- stage: Build_private
+  jobs:
+  - job: build_vsix
+    pool:
+      vmImage: 'ubuntu-latest' 
+    steps:
+    - task: NPM@1
+      displayName: npm install
+      inputs:
+        command: install
+        workingDir: extensions
+
+    - task: NPM@1
+      displayName: npm run lint
+      inputs:
+        command: custom
+        customCommand: run lint
+        workingDir: extensions
+
+    - task: NPM@1
+      displayName: npm run build
+      inputs:
+        command: custom
+        customCommand: run build
+        workingDir: extensions
+
+    - task: CopyFiles@2
+      displayName: Stage extension artifact for publish
+      inputs:
+        sourceFolder: extensions
+        contents: '**/*.vsix'
+        targetFolder: $(Build.ArtifactStagingDirectory)
+
+    - task: PublishPipelineArtifact@0
+      displayName: Publish extension artifact
+      inputs:
+        artifactName: drop
+        targetPath: $(Build.ArtifactStagingDirectory)
+
+    - task: TfxInstaller@2
+      displayName: 'Install Node CLI'
+
+    - task: PublishAzureDevOpsExtension@2
```

Comment:
We could create a single pipeline with different variables configured do reduce code, e.g.:

```
    - task: PublishAzureDevOpsExtension@2
      condition: ne($(nlu.extensionVisibility), 'public')
      inputs:
        ...
        extensionVisibility: $(nlu.extensionVisibility)
        ...
```

Reply:
You could also consider setting up a matrix strategy so we don't need to configure variables on the pipeline:
https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema#matrix

E.g.,
```
strategy:
  matrix: { build_public_extension: { extensionVisibility: 'public' }, build_private_extension: { extensionVisibility: 'private' } }
  maxParallel: 1
```

Reply:
That being said, it's not much code sharing, and it's probably easier to configure the "public" build to only run on merge to master if we don't merge.

Reply:
I don't think we can apply matrix here as it creates 3 different jobs in one build and not two different pipelines for one pipeline file. however I created two separate pipelines for one pipeline file and disable some of the steps using env var in conditions.